### ISSUE TITLE
Layout/EmptyLineAfterGuardClause disallows consecutive guard clauses if they span multiple lines

### ISSUE THREAD

**Issue by dari-us** [rubocop/rubocop#15021]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Layout/EmptyLineAfterGuardClause

### Steps to reproduce the problem

Using lengthy conditions and error message for guard clauses makes Rubocop want to separate them with empty lines.
I am using a method that has the only purpose of validating something and raising in case a condition is not met, so it is basically a long chain of guard clauses.

```ruby
def validate_something
  raise SomeError, "this check has failed" unless short_condition # rubocop complains
  if some_lengthy_condition_does_not_meet_requirements
    raise LengthyError.new("this is some lengthy error message that must be split into multiple lines",
                           because: "i need to add some metadata while satisfying Layout/LineLength cop")
  end
  raise AnotherError, "another check has failed" unless another_check_succeeds # rubocop does not complain
  if another_lengthy_condition_does_not_meet_requirements
    raise ShortError.new("this is a short error that fits into one line and everyone is happy", because: "it is treated as one line")
  end

  return true
end
```

### Expected behavior

The `Layout/EmptyLineAfterGuardClause` cop should not complain because the code in question are two consecutive guard clauses with no additional statements in between them.
So to me, it should still count as a "one-liner", just like the last check, which is *technically* 3 lines, because of `if ...` and `end`, but Rubocop already recognizes this and does not require an empty line.

### Actual behavior

Rubocop complains.
I think it should not complain, because the code in question are two consecutive guard clauses without additional statements in between them.

```
test.rb:2:3: C: [Correctable] Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.
  raise SomeError, "this check has failed" unless short_condition # rubocop complains
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### RuboCop version

```shell
1.85.1 (using Parser 3.3.10.2, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 4.0, running on ruby 4.0.1) [x86_64-linux]
```
